hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -2791,7 +2791,8 @@ public static Option<HoodieIndexVersion> getIndexVersionOption(String metadataPa
     if (!indexDefs.containsKey(metadataPartitionPath)) {
       return Option.empty();
     }
-    return Option.of(indexDefs.get(metadataPartitionPath).getVersion());
+    HoodieIndexVersion version = indexDefs.get(metadataPartitionPath).getVersion();
+    return version != null ? Option.of(version) : Option.empty();
```

Comment:
@linliu-code @rahil-c 

```
public static <T> Option<T> ofNullable(T value) {
    return null == value ? empty() : of(value);
  }
```

does this already.. :)

Reply:
Thanks for the pointer @vinothchandar made the change